homeassistant/components/intellifire/__init__.py

```diff
@@ -31,82 +42,152 @@
 ]
 
 
+def _construct_common_data(entry: ConfigEntry) -> IntelliFireCommonFireplaceData:
+    """Convert a config entry into IntelliFireCommonFireplaceData."""
+
+    return IntelliFireCommonFireplaceData(
+        auth_cookie=entry.data[CONF_AUTH_COOKIE],
+        user_id=entry.data[CONF_USER_ID],
+        web_client_id=entry.data[CONF_WEB_CLIENT_ID],
+        serial=entry.data[CONF_SERIAL],
+        api_key=entry.data[CONF_API_KEY],
+        ip_address=entry.data[CONF_IP_ADDRESS],
+        read_mode=entry.options[CONF_READ_MODE],
+        control_mode=entry.options[CONF_CONTROL_MODE],
+    )
+
+
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    if config_entry.version == 1:
+        new = {**config_entry.data}
+
+        # Rename Host to IP Address
+        new[CONF_IP_ADDRESS] = new.pop("host")
+
+        # Use credentials to get data I guess...
+        serial = config_entry.title.replace("Fireplace ", "")
```

Comment:
People can change the title of the old entry. This isn't a reliable way of fetching serial data

Reply:
But you're logging in here, don't you get serial data from that call?

Reply:
I think I'll go with:

If we can extract serial we try that - if not - try IP otherwise ... I guess the config entry is just invalid because they'll have to start from scratch since we have no way of identifying what the current entry is/was.

(Like if they have swapped routers so DHCP change of IP address, and they renamed the unit.)

Is there an abort I can call for that case or is that too far out?